src/vendors/VendorDetails.tsx

```diff
@@ -60,9 +65,15 @@ const VendorDetails = ({ vendor, close, action }: Props) => {
           )}
           {((location.latitude !== 0 && location.longitude !== 0) || street) && (
             <TouchableOpacity
-              onPress={() =>
-                initiateDirection({ title, coordinate: location, building_number, street, city })
-              }
+              onPress={() => {
+                try {
+                  initiateDirection({ title, coordinate: location, building_number, street, city })
+                } catch (error) {
+                  Logger.warn('Directions', error)
+                  dispatch(showError(ErrorMessages.FAILED_OPEN_DIRECTION))
+                  return
+                }
```

Comment:
This is not very readable try creating a function `handleOpenMap` that wraps line 69 to 75, and line 68 turns to `onPress={handleOpenMap}`.

Reply:
Ill do that and merge!